### PR TITLE
Add demo Function strategy

### LINKED ISSUE #29: Update with a Function strategy
Once EMMC-ASBL/oteapi-core#73 has been merged, a new strategy type is available: Function strategies.

These should be added to the template.

### PR DESCRIPTION
**Pull request by CasperWA**
Closes #29

### CODE REVIEW

File: {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        Parameters:
+            session: A session-specific dictionary context.
+
+        Returns:
+            Dictionary of key/value-pairs to be stored in the sessions-specific
+            dictionary context.
+
+        """
+        return {}
```
Comment (jesper-friis) [inline {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py:34]:
```suggestion
        return SessionUpdate()
```

File: {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
```
Comment (jesper-friis) [inline {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py:50]:
```suggestion
        return SessionUpdate()
```

File: {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+"""Demo function strategy class."""
+# pylint: disable=no-self-use,unused-argument
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
```
Comment (jesper-friis) [inline {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py:4]:
```python
from oteapi.models import SessionUpdate
```

**Review by jesper-friis — COMMENTED** [review]
Looks good. Nowdays I guess we should return a SessionUpdate()...

File: {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py (hunk 1 above)
Comment (CasperWA) [reply {{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/function.py:34]: This (as well as for the other comments) are implemented in #45

**Review by jesper-friis — APPROVED** [review]

**Pull request merged**